src/flinkStatementResultsManager.ts

```diff
@@ -17,22 +18,64 @@ import { parseResults } from "./utils/flinkStatementResults";
 
 const logger = new Logger("flink-statement-results");
 
-type MessageType =
+export type ResultCount = { total: number; filter: number | null };
+export type StreamState = "running" | "completed";
+
+export type MessageType =
   | "GetResults"
   | "GetResultsCount"
   | "GetSchema"
-  | "GetMaxSize"
   | "GetStreamState"
   | "GetStreamError"
   | "PreviewResult"
   | "PreviewAllResults"
   | "Search"
-  | "GetSearchQuery"
   | "SetVisibleColumns"
   | "GetStatementMeta"
   | "StopStatement";
 
-type StreamState = "running" | "completed";
+// Define the post function type based on the overloads
+export type PostFunction = {
```

Comment:
In future, some / all of these would llove some docstrings esp when the functionality implied isn't 100% obvious by the name.

And, oh, a newline between each individual one so isn't as much a wall of noise?

Reply:
Agreed, but hey -- I'm happy that it's now defined in the same file.